processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/ProcessPipeline.java

```diff
@@ -124,6 +125,19 @@ Completion process(TaskRequest request, DecatonTask<T> task) throws InterruptedE
                 new ProcessingContextImpl<>(scope.subscriptionId(), request, task, processors, retryProcessor,
                                             scope.props());
 
+        if (task.metadata().timestampMillis() > 0) {
```

Comment:
Can you put this logic here instead? https://github.com/line/decaton/pull/218/files#diff-4df160cb6c95ec9c8815d2af1dda8a9ebe841c74c4e094bd3ef8e75074dd0af5R100

Just because I think we want to keep this `process` method to have lines that are essential to execute the stage of executing `DecatonProcessor#process`.